### ISSUE TITLE
Crash in RxPermissionsFragment

### ISSUE THREAD

**Issue by ajalt** [tbruyelle/RxPermissions#106]
I've been seeing some crash reports caused by RxPermissions:

```
Fatal Exception: java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:android:fragment:0, request=42, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS VirtualScreenParam=Params{mDisplayId=-1, null, mFlags=0x00000000)} (has extras) }} to activity {com.getkeepsafe.applock/com.getkeepsafe.applock.ui.signup.ResetPasswordActivity}: java.lang.IllegalStateException: RxPermissions.onRequestPermissionsResult invoked but didn't find the corresponding permission request.
       at android.app.ActivityThread.deliverResults(ActivityThread.java:4925)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:4968)
       at android.app.ActivityThread.access$1600(ActivityThread.java:222)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1849)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by java.lang.IllegalStateException: RxPermissions.onRequestPermissionsResult invoked but didn't find the corresponding permission request.
       at com.tbruyelle.rxpermissions.RxPermissionsFragment.onRequestPermissionsResult(RxPermissionsFragment.java:61)
       at com.tbruyelle.rxpermissions.RxPermissionsFragment.onRequestPermissionsResult(RxPermissionsFragment.java:51)
       at android.app.Activity.dispatchRequestPermissionsResultToFragment(Activity.java:7273)
       at android.app.Activity.dispatchActivityResult(Activity.java:7145)
       at android.app.ActivityThread.deliverResults(ActivityThread.java:4921)
       at android.app.ActivityThread.handleSendResult(ActivityThread.java:4968)
       at android.app.ActivityThread.access$1600(ActivityThread.java:222)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1849)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7229)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

I've never been able to reproduce this locally. I don't create any permission requests outside of RxPermissions. This is an uncommon crash, occurring in less than 1 in 1,000 requests. Maybe the process is destroyed and recreated before the result is delivered? 

This crash isn't required, correct? I understand that it can be nice for development, but could you add and option to ignore unknown permissions requests?

**Comment by tbruyelle** [comment]
Well this bug is hard-boiled. We had it in the past and did ugly fixes to remove it. I really though it won't come back since we use a fragment and not an activity. But the Android framework is definitively not a good friend. 

I'll commit a fix to ignore the crash, because now I'm tired.

**Comment by tbruyelle** [comment]
Fixed in version 0.9.1
